pkg/resources/utils.go

```diff
@@ -694,6 +715,48 @@ func IngressForLegacyUI(instance *operatorsv1alpha1.LegacyHeader) *netv1.Ingress
 	return ingress
 }
 
+func BuildCertificate(instanceNamespace, instanceClusterIssuer string, certData CertificateData) *certmgr.Certificate {
+	reqLogger := log.WithValues("func", "BuildCertificate")
+
+	metaLabels := labelsForCertificateMeta(certData.App, certData.Component)
+	var clusterIssuer string
+	if instanceClusterIssuer != "" {
+		reqLogger.Info("clusterIssuer=" + instanceClusterIssuer)
+		clusterIssuer = instanceClusterIssuer
+	} else {
+		reqLogger.Info("clusterIssuer is blank, default=" + DefaultClusterIssuer)
+		clusterIssuer = DefaultClusterIssuer
+	}
+
+	certificate := &certmgr.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      certData.Name,
+			Labels:    metaLabels,
+			Namespace: instanceNamespace,
+		},
+		Spec: certmgr.CertificateSpec{
+			CommonName: certData.Common,
+			SecretName: certData.Secret,
+			IsCA:       false,
+			DNSNames: []string{
+				certData.Common,
+				certData.Common + "." + instanceNamespace,
+				certData.Common + "." + instanceNamespace + ".svc.cluster.local",
+			},
+			Organization: []string{"IBM"},
+			IssuerRef: certmgr.ObjectReference{
+				Name: clusterIssuer,
+				Kind: certmgr.ClusterIssuerKind,
+			},
+		},
+	}
+	return certificate
+}
+
+func labelsForCertificateMeta(appName, componentName string) map[string]string {
+	return map[string]string{"app": appName, "component": componentName, "release": ReleaseName}
```

Comment:
let's add a few more labels, better to track for the operator
`app.kubernetes.io/instance=ibm-commonui-operator`
`app.kubernetes.io/managed-by=ibm-commonui=operator`
`app.kubernetes.io/name=name of certificate`